### PR TITLE
Add credentions page, rewok try/catch in methods

### PR DESCRIPTION
**Pull request by MariyaSin**
http://outsourcing.nat.tepkom.ru:20080/redmine/issues/659

### CODE REVIEW

File: src/GUI/App.razor [hunk 1]
```diff
@@ -16,6 +16,9 @@
 @code {
     protected override void OnInitialized()
     {
-        UriHelper.NavigateTo("");
+        if(UriHelper.Uri.IndexOf("credentials") == -1)
```
Comment (djino97) [inline src/GUI/App.razor]: add empty space

File: src/GUI/Pages/Admin/User/CreateUser.razor.cs [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+            }
+            catch(Services.ApiClients.UserService.ApiException<Services.ApiClients.UserService.ErrorResponse> ex)
+            {
+                _message = ex.Result.Message;
+            }
+            catch(Exception ex)
+            {
+                //remove when spec reworked
+                _message = ex.Message;
+            }
```
Comment (djino97) [inline src/GUI/Pages/Admin/User/CreateUser.razor.cs]: nee remove

File: src/GUI/Pages/Admin/User/CreateUser.razor.cs [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+        private ElementReference _statusSelect;
+
+        public CreateUser()
+        {
+            _userCommunication.Type = CommunicationType.Email;
+            _userData.Communications = new List<CommunicationInfo>();
+            _userData.Communications.Add(_userCommunication);
+        }
+
+        private async Task GeneratePassword()
```
Comment (djino97) [inline src/GUI/Pages/Admin/User/CreateUser.razor.cs]: add postfix Async

File: src/GUI/Pages/Admin/User/CreateUser.razor.cs [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+        {
+            _userData.Password = await userService.GeneratePassword();
+        }
+
+        private async Task HandleValidSubmit()
```
Comment (djino97) [inline src/GUI/Pages/Admin/User/CreateUser.razor.cs]: add postfix Async

File: src/GUI/Pages/Public/Auth.razor.cs [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+namespace LT.DigitalOffice.GUI.Pages.Public
+{
+    public partial class Auth
+    {
+        private AuthenticationRequest _authData = new();
+        private string _message;
+
+        private ElementReference _passwordInput;
+
+        private async Task HandleValidSubmit()
```
Comment (djino97) [inline src/GUI/Pages/Public/Auth.razor.cs]: add postfix Async

File: src/GUI/Pages/Public/CreateCredentials.razor.cs [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+
+        private ElementReference _passwordInput;
+
+        protected async override void OnInitialized()
+        {
+            _credentialsData.UserId = Id;
+            StateHasChanged();
+        }
+
+        private async Task HandleValidSubmit()
```
Comment (djino97) [inline src/GUI/Pages/Public/CreateCredentials.razor.cs]: add postfix Async

File: src/GUI/Services/AuthService.cs [hunk 7]
```diff
@@ -11,6 +11,7 @@ namespace LT.DigitalOffice.GUI.Services
 {
     public class AuthService : IAuthService
     {
+        private readonly HttpClient _client = new();
```
Comment (djino97) [inline src/GUI/Services/AuthService.cs]: move initialization in constructor

File: src/GUI/Services/AuthService.cs [hunk 8]
```diff
@@ -20,26 +21,15 @@ public AuthService(AuthenticationStateProvider provider, ISessionStorageService
             _storage = storage;
         }
 
-        public async Task<string> Login(AuthenticationRequest request)
+        public async Task Login(AuthenticationRequest request)
```
Comment (djino97) [inline src/GUI/Services/AuthService.cs]: add postfix Async

**Review by djino97 — APPROVED** [review]

File: src/GUI/App.razor (hunk 1 above)
Comment (MariyaSin) [reply src/GUI/App.razor]: done

File: src/GUI/Pages/Admin/User/CreateUser.razor.cs (hunk 3 above)
Comment (MariyaSin) [reply src/GUI/Pages/Admin/User/CreateUser.razor.cs]: done

File: src/GUI/Pages/Admin/User/CreateUser.razor.cs (hunk 4 above)
Comment (MariyaSin) [reply src/GUI/Pages/Admin/User/CreateUser.razor.cs]: done

File: src/GUI/Pages/Public/Auth.razor.cs (hunk 5 above)
Comment (MariyaSin) [reply src/GUI/Pages/Public/Auth.razor.cs]: done

File: src/GUI/Pages/Public/CreateCredentials.razor.cs (hunk 6 above)
Comment (MariyaSin) [reply src/GUI/Pages/Public/CreateCredentials.razor.cs]: done

File: src/GUI/Services/AuthService.cs (hunk 7 above)
Comment (MariyaSin) [reply src/GUI/Services/AuthService.cs]: done

File: src/GUI/Services/AuthService.cs (hunk 8 above)
Comment (MariyaSin) [reply src/GUI/Services/AuthService.cs]: done

**Review by TIove — APPROVED** [review]

File: src/GUI/Pages/Auth/CreateCredentials.razor.cs [hunk 9]
```diff
@@ -0,0 +1,56 @@
...
+            catch (ApiException<ErrorResponse> ex)
+            {
+                _message = ex.Result.Message;
+            }
+            catch (ApiException<OperationResultResponse> ex)
+            {
+                foreach (var error in ex.Result.Errors)
+                {
+                    _message = _message + " " + error;
+                }
```
Comment (djino97) [inline src/GUI/Pages/Auth/CreateCredentials.razor.cs]: use string.Join

File: src/GUI/Pages/Admin/User/CreateUser.razor.cs [hunk 10]
```diff
@@ -0,0 +1,67 @@
...
+        {
+            _userData.Password = await userService.GeneratePasswordAsync();
+        }
+
+        private async Task HandleValidSubmitAsync()
+        {
+            try
+            {
+                await userService.CreateUserAsync(_userData);
+                UriHelper.NavigateTo("/");
```
Comment (djino97) [inline src/GUI/Pages/Admin/User/CreateUser.razor.cs:42]: Why use redirect after create user?

File: src/GUI/Services/CompanyService.cs [hunk 11]
```diff
@@ -4,90 +4,54 @@
...
-                return null;
-            }
+            return await _client.FindPositionsAsync(token, 0, 100, false);
+        }
+
+        public async Task<FindResultResponseOfficeInfo> FindOfficesAsync()
+        {
+            var token = await _storage.GetItemAsync<string>(Consts.AccessToken);
+
+            return await _client.FindOfficesAsync(token, 0, 100, null);
```
Comment (djino97) [inline src/GUI/Services/CompanyService.cs]: takeCount = int.MaxValue

Comment (MariyaSin) [reply src/GUI/Services/CompanyService.cs]: done

File: src/GUI/Pages/Admin/User/CreateUser.razor.cs (hunk 10 above)
Comment (MariyaSin) [reply src/GUI/Pages/Admin/User/CreateUser.razor.cs:42]: to close modal window

**Review by djino97 — APPROVED** [review]

**Review by borovHokage — APPROVED** [review]

**Review by BelehovEgor — APPROVED** [review]

**Pull request merged**